distributed/scheduler.py

```diff
@@ -3965,6 +3965,8 @@ def transition(self, key, finish, *args, **kwargs):
                     self.tasks[ts.key] = ts
                 for plugin in list(self.plugins):
                     try:
+                        if self.tasks[key].processing_on is not None:
+                            kwargs['worker'] = self.tasks[key].processing_on.address
```

Comment:
Thanks @nicolls1 !

Perhaps instead we might just place in the entire task state object, and let the plugin do what it wants with it.  Thoughts?

Reply:
To be clear, I'm suggesting something like `kwargs['task_state'] = ts`

Reply:
Thanks for the quick response. That sounds much cleaner and more future proof, will do!